Code/Source/RobotControl/RobotControl.h

```diff
@@ -26,15 +31,22 @@ namespace ROS2
     class RobotControl : public IRobotControl
     {
     public:
-        // TODO - pass component args (qos, topic name, etc) as editor-enabled serializable struct
-        void Activate(const AZ::Entity* entity, const AZStd::string& topicName, const QoS &qos) final
+        void Activate(const AZ::Entity* entity,
+                      const ControlConfiguration& controlConfiguration) final
         {
-            SetTargetComponent(entity);
+            m_controlConfiguration = controlConfiguration;
             m_active = true;
             if (!m_controlSubscription)
             {
+                auto ros2Frame = entity->FindComponent<ROS2FrameComponent>();
+                auto namespacedTopic = ROS2Names::GetNamespacedName(
```

Comment:
minor: 
according to o3de guidelines, avoid "auto" unless type is obvious / irrelevant. Here it is not clear what kind of string namespacedTopic is (ros2Frame type is obvious though)